packages/relay/tests/lib/precheck.spec.ts

```diff
@@ -427,4 +427,37 @@ describe('Precheck', async function () {
+    const contractCall = '0xcfae3217';
+    const transfer = '0x';
+    const invalidTx = '0x60806040523480156200001157600080fd5b';
+    it('should be able to calculate contract create', function () {
+      const intrinsicGasCost = Precheck.transactionIntrinsicGasCost(contractCreate);
+      expect(intrinsicGasCost).to.be.greaterThan(constants.TX_BASE_COST);
+    });
+
+    it('should be able to calculate contract call', function () {
+      const intrinsicGasCost = Precheck.transactionIntrinsicGasCost(contractCall);
+      expect(intrinsicGasCost).to.be.greaterThan(constants.TX_BASE_COST);
+    });
+
+    it('should be able to calucate tx without starting 0x', function () {
+      const contractCallTrimmed = contractCall.replace('0x', '');
+      const intrinsicGasCost = Precheck.transactionIntrinsicGasCost(contractCallTrimmed);
+      expect(intrinsicGasCost).to.be.greaterThan(constants.TX_BASE_COST);
+    });
+
+    it('should be able to able to calculate transfer', function () {
+      const intrinsicGasCost = Precheck.transactionIntrinsicGasCost(transfer);
+      expect(intrinsicGasCost).to.be.equal(constants.TX_BASE_COST);
+    });
+
+    it('should be able to calculate for odd length tx', function () {
+      const intrinsicGasCost = Precheck.transactionIntrinsicGasCost(invalidTx);
+      expect(intrinsicGasCost).to.be.greaterThan(constants.TX_BASE_COST);
+    });
+  });
```

Comment:
Can you add some more fine tuned examples that confirm exact numbers for a a small and medium set of contract bytes e.g.
- empty contract (smallest permissable)
- simple contract with a singleslot that is set on constructor
- you can also use existing test contracts in the repo

Doesn't have to be exhaustive but 1 or 2 to know the number calculation work for the intrinsic gas calculation on a create and not on create scenario

Reply:
@georgi-l95 can address this in follow up